### PR TITLE
Closes #577 i18n support

### LINKED ISSUE #577: Add localization support to GitLens
Here is a [sample extension](https://github.com/Microsoft/vscode-extension-samples/tree/master/i18n-sample) using the [vscode-nls](https://github.com/Microsoft/vscode-nls) library

### PR DESCRIPTION
**Pull request by airaketa**
# Description

Closes #577 
Implemented i18n using vscode-nls and vscode-nls-dev

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes include any required corresponding changes to the documentation
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

**Comment by airaketa** [comment]
@eamodio can you take a look at this? Ready to make edits to get it merged asap

**Comment by eamodio** [comment]
@airaketa Wow! This is awesome! It will definitely take me some time to review all of the changes.

I did notice that some strings were changed, though -- could I ask you to check to ensure that no strings have changed in this process, unless there is a specific reason to change one.

**Comment by airaketa** [comment]
@eamodio One of the reasons that I changed some  strings was that in some languages word's suffix depends on its context in a sentence.

Also I had to remove `pluralize` function calls because some languages have 3 plural forms instead of 2 like it is in English (e.g. slavic languages where you have different word suffix for amount of 1, 2-4 and >4 accordingly).
In this case I preserved 2 plural forms like it is in English to keep code as simple as possible, but the whole sentence substitution allows localizators to change word order to make it sound authentic in the target language.
E.g. one can write:
1. ветка впереди на 1 коммит (branch is 1 commit ahead)
2. ветка впереди на несколько ({0}) коммитов (branch is several ({0}) commits ahead)

Instead of 3 forms:
1. ветка впереди на 1 коммит (branch is 1 commit ahead)
2. ветка впереди на 3 коммит**а** (branch is 3 commits ahead)
3. ветка впередин на 5 коммит**ов** (branch is 5 commits ahead)

In other words, the idea was to replace "composite" strings, which are assembled from different combinations of words and their forms, with completed sentences so that localizators could decide which form should one or another word have depending on the context.
I believe this answer can help to understand this principles better: https://stackoverflow.com/a/390247

So there should be no difference for English-speaking users, feel free to tell if there is one. I will also try to fix such issues if I notice them.

**Comment by Cyberhan123** [comment]
Is there anything new and can I help

**Comment by enhongchen** [comment]
@eamodio 催一催进度😂

**Comment by zhang-wenchao** [comment]
@eamodio Is there any progress in this?

**Comment by zhang-wenchao** [comment]
@eamodio Please try to pay attention to this issue.

**Comment by eamodio** [comment]
Closing as this has now been added, see: https://github.com/gitkraken/vscode-gitlens/issues/577#issuecomment-5610659116
